afk-daily.sh

```diff
@@ -1,1610 +1,1850 @@
-        input tap 60 1850 # Return
-        wait
+        inputTapSleep 60 1850                   # Return
 
-        # Check for confirm to exit button
-        getColor 715 1260
-        if [ "$RGB" == "feffff" ]; then
-            input tap 715 1260 # Confirm
-            wait
-        fi
+        testColorORTapSleep 715 1260 feffff     # Check for confirm to exit button
     else
         # Quick exit battle
-        input tap 550 1850 # Battle
-        sleep 1
-        input tap 80 1460 # Pause
-        wait
-        input tap 230 960 # Exit
-        sleep 1
+        inputTapSleep 550 1850 1                # Battle
+        inputTapSleep 80 1460                   # Pause
+        inputTapSleep 230 960 1                 # Exit
 
-        # Check for multi-battle
-        getColor 450 1775
-        if [ "$RGB" != "cc9261" ]; then
-            input tap 70 1810
+        if testColorNAND 450 1775 cc9261; then  # Check for multi-battle
+            inputTapSleep 70 1810
         fi
     fi
 
     wait
     verifyRGB 450 1775 cc9261 "Challenged boss in campaign." "Failed to fight boss in Campaign."
 }
 
-# Collects fast rewards
-function fastRewards() {
-    input tap 950 1660
-    sleep 1
-    input tap 710 1260
-    sleep 2
-    input tap 560 1800
-    sleep 1
-    input tap 400 1250
+# ##############################################################################
+# Function Name : collectFriendsAndMercenaries
+# Descripton    : Collects and sends companion points, as well as auto lending mercenaries
+# ##############################################################################
+collectFriendsAndMercenaries() {
+    if [ $DEBUG -ge 4 ]; then echo "[DEBUG] collectFriendsAndMercenaries" >&2; fi
+    inputTapSleep 970 810 1                     # Friends
+    inputTapSleep 930 1600                      # Send & Recieve
+    if testColorOR 825 1750 df1909; then        # Check if its necessary to send mercenaries
+        inputTapSleep 720 1760                  # Sort-Term
+        inputTapSleep 990 190                   # Manage
+        inputTapSleep 630 1590                  # Apply
+        inputTapSleep 750 1410 1                # Auto Lend
+        inputTapSleep 70 1810 0                 # Return
+    else
+        echo "[INFO] No mercenaries to lend..."
+    fi
+    inputTapSleep 70 1810 0                     # Return
 
     wait
+    verifyRGB 450 1775 cc9261 "Sent and recieved companion points, as well as auto lending mercenaries." "Failed to collect/send companion points or failed to auto lend mercenaries."
+}
+
+# ##############################################################################
+# Function Name : fastRewards
+# Descripton    : Collects fast rewards
+# ##############################################################################
+fastRewards() {
+    if [ $DEBUG -ge 4 ]; then echo "[DEBUG] fastRewards" >&2; fi
+    if testColorOR 980 1620 ef1e05; then        # check red dot to see if free fast reward is avaible
```

Comment:
Is this talking about [this](https://github.com/zebscripts/AFK-Daily/pull/59#issue-623663486)? I'm guessing it's implemented, so why isn't it marked as done?

Reply:
Yes, my bad, forgot to validate it.